account-kit/smart-contracts/src/6900-v08-RI/modules/single-signer-validation/module.ts

```diff
@@ -10,8 +12,10 @@ const meta = {
   addresses,
 };
 
+// Todo: some unified type for ERC-6900 v0.8 modules. I couldn't figure out how to parameterize the class itself over the abi type parameters for onInstall and onUninstall.
```

Comment:
can you explain this? what's your ideal devex?

Reply:
Ah this is just a todo for later, but basically it'd be nice to have a base "class" or other util for creating module definitions that just takes in:
- An address record
- Name / meta info
- The parameter formats for `onInstall` / `onUninstall`

And it could produce something that fits the interface I created and have been using so far. I don't think it needs to be as heavyweight as the old plugingen, since most config formats are now off-chain rather than in the manifest, but it would be nice to have an enforced interface / type for the modules.